orchestra/models.py

```diff
@@ -14,6 +12,56 @@
 # TODO(marcua): Convert ManyToManyFields to django-hstore referencefields or
 # wait for django-postgres ArrayFields in Django 1.8.
 
+class Workflow(models.Model):
+    """
+    Workflows describe the steps and requirements of completing expert work.
+
+    This model represents workflows that have been loaded into the database and
+    are usable by Orchestra projects. All workflows are also defined in the
+    codebase (see the `root_directory` attribute).
+
+    Attributes:
+        slug (str):
+            Unique identifier for the workflow.
+        name (str):
+            Human-readable name for the workflow.
+        description (str):
+            A longer description of the workflow.
+        code_directory (str):
+            The full path to the location of the workflow's manifest.
+    """
+    slug = models.CharField(max_length=200, unique=True)
+    name = models.CharField(max_length=200)
+    description = models.TextField()
+    code_directory = models.CharField(max_length=255, unique=True)
+
+    class Meta:
+        app_label = 'orchestra'
+
+
+class WorkflowVersion(models.Model):
+    """
+    WorkflowVersions represent changes made to a single workflow over time.
+
+    Attributes:
+        slug (str):
+            Unique identifier for the workflow version.
+        name (str):
+            Human-readable name for the workflow version.
+        description (str):
+            A longer description of the workflow version.
+        workflow (orchestra.models.Workflow):
+            The workflow that this is a version of.
+    """
+    slug = models.CharField(max_length=200)
```

Comment:
Right now there is no assumption made on the slug. The semantics are, 'A version is any direct subdirectory of a workflow's code_directory containing a valid `version.json`'. This lets us just search all subdirectories of the workflow for `version.json` files, and discover each version's slug. I'd be fine adding a code_directory for the version, but not all code for the version is required to be in the directory (e.g., static assets shared across versions), so it feels a bit odd.